src/main/java/org/openrewrite/java/spring/cloud2022/MigrateRequestMappingOnFeignClient.java

```diff
@@ -60,28 +52,29 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
-                        List<J.Annotation> annotations = service(AnnotationService.class).getAllAnnotations(getCursor());
-                        J.Annotation requestMapping = annotations.stream()
-                            .filter(a -> TypeUtils.isOfClassType(a.getType(), REQUEST_MAPPING))
-                            .findFirst().orElse(null);
-                        J.Annotation feignClient = annotations.stream()
-                            .filter(a -> TypeUtils.isOfClassType(a.getType(), FEIGN_CLIENT))
-                            .findFirst().orElse(null);
+                    J.Annotation requestMapping = classDecl.getLeadingAnnotations().stream()
+                        .filter(a -> TypeUtils.isOfClassType(a.getType(), REQUEST_MAPPING))
+                        .findFirst().orElse(null);
+                    J.Annotation feignClient = classDecl.getLeadingAnnotations().stream()
+                        .filter(a -> TypeUtils.isOfClassType(a.getType(), FEIGN_CLIENT))
+                        .findFirst().orElse(null);
 
-                        if (requestMapping != null && feignClient != null) {
-                            J.ClassDeclaration cd = classDecl;
-                            if (requestMapping.getArguments() == null || requestMapping.getArguments().isEmpty()) {
-                                cd = removeRequestMapping(classDecl);
-                            } else if (requestMapping.getArguments().size() == 1) {
-                                J.Assignment path = convertToPathAttribute(requestMapping.getArguments().get(0));
-                                if (path != null && !hasPathAttribute(feignClient)) {
-                                    cd = addAttributeToFeignClient(removeRequestMapping(classDecl), path);
-                                }
+                    if (requestMapping != null && feignClient != null) {
+                        J.ClassDeclaration cd = classDecl;
+                        if (requestMapping.getArguments() == null || requestMapping.getArguments().isEmpty()) {
+                            cd = removeRequestMapping(cd, ctx);
+                        } else if (requestMapping.getArguments().size() == 1) {
+                            String pathValueFromRequestMapping = getPathValue(requestMapping.getArguments().get(0));
+                            if (pathValueFromRequestMapping != null && !hasPathAttribute(feignClient)) {
+                                cd = removeRequestMapping(classDecl, ctx);
+                                cd = classDecl.withLeadingAnnotations(
+                                    ListUtils.map(cd.getLeadingAnnotations(), a -> (J.Annotation)
+                                        new AddOrUpdateAnnotationAttribute(FEIGN_CLIENT, "path",
+                                            pathValueFromRequestMapping, true, false).getVisitor()
+                                            .visit(a, ctx, getCursor().getParentOrThrow())));
```

Comment:
Thanks for your quick adaption, look good now! 
One last thought: I feel that it would round up this contribution if this block were inside a method. 
Maybe the unused `addAttributeToFeignClient` one?

Reply:
I think from there we are good to go 🚀